### DISCUSSION TITLE
calcParent2Child

### DISCUSSION THREAD

**Discussion by ArianeNeves** [mtex-toolbox/mtex#910] [Ask Anything]
Hello,
The OR of the material showed a spread around the average OR. 
The calcParent2Child gives the optimized OR, which is the average OR.

In the calcParent2Child function, p2cCandidates means the orientation relationship of an iteration considering the child2child misorientation?

**Comment by ralfHielscher** [comment]
I'm not sure what kind of answer you expect :) It's sort of a gradient towards the optimized OR. 
Explaining this correctly can not be done withing this discussion thread.
May I ask why this is of interest for you?

Ralf.

**Comment by ArianeNeves** [comment]
Ralf, 
Excuse me. I have tried to plot the experimental OR in the Euler space in order to show the spread of the OR around the optimized OR as reported in this paper https://doi.org/10.1016/j.jnucmat.2019.01.022. The paper's approach is different. The determination of OR is carried out based on the misorientation between austenite and martensite.

Based on your explanation, the spread of the OR around the optimized OR is stored in p2cCandidates?

Can you understand me?

**Comment by nyyssont** [comment]
I understand that you would be interested in all of the various candidates for a representative average orientation relationship, which is calculated (usually) from the misorientations between neighboring child grain mean orientations.

Based on calcParent2Child code, this data is indeed given by p2cCandidates. If you are interested in this output, you could possibly change the first line of the function to:
```
function [p2c, omega, p2cCandidates] = calcParent2Child(mori,p2c,varargin)
```

The following command will then give you the candidates:
```
[p2c, omega, p2cCandidates] = calcParent2Child(mori,p2c)
```

You should note that you will have to work outside the job class and construct the misorientation object yourself from grain pairs (or grain boundary misorientations etc.) to obtain p2cCandidates.

You might also want to consider what is the best way to observe the "spread" in orientation relationship. An individual misorientation may, after all, result in a misindexation and an erroneous OR. A good way would perhaps be to fully reconstruct the prior austenite structure and then compare the child and parent orientations directly:
```
mori = inv(childOri) .* parenOri;
```
I would also like to recommend Dr. Niessen's and Dr. Gazder's orientation relationship tool discovery add-on for MTEX. It can be found here:
https://github.com/frankNiessen/ORTools

There is a rather helpful array of introductory videos to get you started. Hope this helps!

**Reply by ralfHielscher** [reply]
Hi Tuomo,

I think one needs to be careful here:

1. p2cCandidates is not the "true" OR for the misorientation `mori`. It simply does not satisfy `mori = p2cCandidate * inv(p2cCandidate)`.
2. When fully reconstructing the parent orientations on EBSD level the OR is exactly the assumed orientation relationship. So no spread can be observed here.

Ralf.

**Reply by nyyssont** [reply]
Thank you for the clarification Ralf.

Some rumination on the martensitic transformation in steel follows.

An assumption could be made that in the case of the transformation of recrystallized austenite to martensite, the starting microstructure consists of mostly undeformed austenite. Then the mean orientation of a complete reconstructed austenite grain would be a more meaningful description of the "starting" orientation than an individual reconstructed orientation pixel (which is, after all, simply a rotation using the representative average orientation relationship).

So to get some idea on how the orientation relationship varies across measured orientation map it might make sense to compare the individual martensite orientations and the orientations of complete reconstructed austenite grains.

Using the demo dataset, one would proceed in this manner:

```
% load the data
mtexdata martensite 
plotx2east

% grain reconstruction
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',3*degree);
ebsd(grains(grains.grainSize < 3)).rotations = quaternion.nan;
ebsd(grains(grains.grainSize < 3)).phase = 0;
ebsd(grains(grains.grainSize < 3)).grainId = 0;
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',3*degree);
%%
job = parentGrainReconstructor(ebsd,grains) % define job
job.calcParent2Child; % get representative OR

%%
[fit,c2cPairs] = job.calcGBFit;
[gB,pairId] = job.grains.boundary.selectByGrainId(c2cPairs);

%%
job.calcGBVotes % compute votes from child grain boundaries
job.calcParentFromVote('minVotes',4)

%%
for l = 1:2
job.calcGBVotes('noC2C') % compute votes ONLY from parent-child boundaries;
job.calcParentFromVote('minFit',5*degree) %compute parent orientations from votes
end

%%

parentEBSD = job.calcParentEBSD;
parentEBSD.grainId = 0;
[grains_aus,parentEBSD('Iron fcc').grainId] = calcGrains(parentEBSD('Iron fcc'),'angle',3*degree)
ebsd.grainId = parentEBSD.grainId

mori_p2cs = inv(ebsd(ebsd.grainId ~= 0).orientations).*grains_aus(ebsd(ebsd.grainId ~= 0).grainId).meanOrientation;
```

```mori_p2cs``` would then contain all the misorientations for some analysis of "spread":
```
angle(mean(mori_p2cs),job.p2c)/degree

std(mori_p2cs)/degree
```

One could even observe how the orientation relationship deviates from the mean locally:
```
figure
plot(ebsd(ebsd.grainId ~= 0),angle(mori_p2cs,mean(mori_p2cs))/degree)
caxis([0 3])
colorbar
hold on
plot(grains_aus.boundary,'linewidth',2)
```
![all_ORs](https://user-images.githubusercontent.com/10060198/112724182-51d48e80-8f12-11eb-893b-ee05af570a2d.jpg)

This might be a way to start. In reality, the situation concerning the transformation taking place in the location of each individual measured orientation pixel is not quite as simple, as each stage of transformation introduces some amount of deformation in the austenite structure. During the last stages of transformation, the orientation of the austenite may have changed due to deformation; then again, the orientation relationship may be different for deformed austenite.
